server/src/test/java/de/pnp/manager/utils/TestUpgradeBuilder.java

```diff
@@ -58,6 +102,18 @@ public TestUpgradeBuilder addEffect(UpgradeEffect effect) {
      * Builds an {@link Upgrade} matching this builder.
      */
     public Upgrade build() {
+        if (effects.isEmpty()) {
+            effects.add(SimpleUpgradeEffect.create("default effect"));
+        }
         return new Upgrade(null, name, type, slots, vendorPrice, effects);
     }
+
+    /**
+     * Builds an {@link Upgrade} matching this builder.
+     * <p>
+     * The upgrade is already persisted in the database.
+     */
+    public Upgrade buildPersisted() {
+        return upgradeRepository.insert(universe, build());
+    }
```

Comment:
should be a `persist()` builder call which flips a flag and `build()` will then persist it.